### PR TITLE
Improvements in single qubit state tomography

### PR DESCRIPTION
**Pull request by andrea-pasquale**
Small improvements for single qubit tomography including:

- comparison is now performed with the target density matrix without shots noise
- circuit is stored in qasm

Checklist:
- [x] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.
- [x] Compatibility with Qibo modules (Please edit this section if the current pull request is not compatible with the following branches).
    - [x] Qibo: `master`
    - [x] Qibolab: `main`
    - [x] Qibolab_platforms_qrc: `main`

### CODE REVIEW

**Comment by alecandido** [comment]
> circuit is stored in qasm

I would suggest you to store in the "native" Qibo format, also used for the cloud support. I.e. replace `circuit.to_qasm()` with `json.dumps(circuit.raw)`.
It's simpler (thus less error-prone), and in general it will support more Qibo features than QASM.

File: src/qibocal/protocols/state_tomography.py [hunk 1]
```diff
@@ -140,30 +143,28 @@ def _fit(data: StateTomographyData) -> StateTomographyResults:
...
+        x_exp = 1 - 2 * np.mean(data[qubit, "X"].samples)
+        y_exp = 1 - 2 * np.mean(data[qubit, "Y"].samples)
+        z_exp = 1 - 2 * np.mean(data[qubit, "Z"].samples)
+        measured_density_matrix = 0.5 * (
             matrices.I + matrices.X * x_exp + matrices.Y * y_exp + matrices.Z * z_exp
         )
-        measured_density_matrix_real[qubit] = np.real(density_matrix).tolist()
-        measured_density_matrix_imag[qubit] = np.imag(density_matrix).tolist()
+        measured_density_matrix_real[qubit] = np.real(measured_density_matrix).tolist()
+        measured_density_matrix_imag[qubit] = np.imag(measured_density_matrix).tolist()
```
Comment (alecandido) [inline src/qibocal/protocols/state_tomography.py:162]: Mainly curiosity: why do you need to split real and imaginary parts?

File: src/qibocal/protocols/state_tomography.py [hunk 2]
```diff
@@ -140,30 +143,28 @@ def _fit(data: StateTomographyData) -> StateTomographyResults:
...
     fid = {}
-    for qubit in data.qubits:
-        x_exp, y_exp, z_exp = (
-            1 - 2 * np.mean(data[qubit, basis].samples) for basis in BASIS
+    circuit = data.circuit
+    circuit.density_matrix = True
+    total_density_matrix = NumpyBackend().execute_circuit(circuit=circuit).state()
+    for i, qubit in enumerate(data.targets):
+        traced_qubits = list(range(len(data.qubits)))
+        traced_qubits.remove(i)
```
Comment (alecandido) [inline src/qibocal/protocols/state_tomography.py:151]:
```suggestion
        traced_qubits = [q for q in range(len(data.qubits)) if q != i]
```
slightly more readable...

**Review by alecandido — APPROVED** [review]

File: src/qibocal/protocols/state_tomography.py (hunk 1 above)
Comment (alecandido) [reply src/qibocal/protocols/state_tomography.py:161]: Ok, I will soon add support to store arrays as bytes

**Review by Edoardo-Pedicillo — APPROVED** [review]

**Pull request merged**